### PR TITLE
feat(EC-1950): bring test_attestation to feature parity with test package

### PR DESCRIPTION
**Pull request by robnester-rh**
## Summary

Brings the `test_attestation` package to feature parity with the `test` package so teams adopting OCI-referrer test result attestations get the same policy rigor as those using embedded `TEST_OUTPUT` task results.

- **AC-1**: Configurable result values via rule data (predicate spec vocabulary: PASSED/FAILED/WARNED/ERROR/SKIPPED)
- **AC-2**: `no_erred_test_attestations` deny rule for ERROR results
- **AC-3**: `no_skipped_test_attestations` deny rule for SKIPPED results
- **AC-4**: Informative test support — failed informative tests warn instead of deny
- **AC-5**: Subject digest validation — deny if attestation subject doesn't match the image being evaluated (ANY-match semantics)
- **AC-6**: Rule data schema validation for all 6 new keys
- **AC-7**: 9 new tests covering all ACs

## Predicate alignment

Updated to match the actual `attest-test-result` step action output which uses integer count fields (`failures`, `warnings`, `successes`) rather than string arrays (`failedTests`, `warnedTests`). The `_has_result` helper checks both `predicate.result` (string match) and count fields (`predicate[key] > 0`), following `test.rego`'s `_did_result` pattern.

## Test plan

- [x] 996/996 tests pass (`make quiet-test`)
- [x] Regal lint clean
- [ ] `make conventions-check` (needs CI)
- [ ] Review predicate format against step action with @joejstuart

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by robnester-rh** [comment]
## Fullsend review findings — responses

### Medium

- **[breaking-api] Message format change** — Intentional. Aligned with the actual `attest-test-result` step action output which produces integer counts (`failures`, `warnings`), not string arrays (`failedTests`, `warnedTests`). Will document in release notes.

- **[breaking-schema] Predicate data model change** — Intentional. The previous schema (`failedTests`/`warnedTests` arrays) came from a PoC spec example, not from the actual step action. Enforcement is preserved via dual-path `_has_result` (checks both `predicate.result` string and count fields).

- **[breaking-api] ERROR/SKIPPED violation codes** — Intentional per AC-2/AC-3. Operators now get specific `short_name`s to waive (`no_erred_test_attestations`, `no_skipped_test_attestations`) instead of the generic `test_result_known`.

- **[breaking-api] subject_mismatch is net-new enforcement** — Intentional per AC-5. Closes a security gap where a valid-but-mismatched attestation (produced for image A but attached to image B) would pass. Will discuss phased rollout via `effective_on` with @joejstuart.

- **[breaking-api] subject_digest moved** — Moved from `slsa_build_scripted_build` to `intoto.rego` to avoid reimplementing digest format conversion. The `slsa_build_scripted_build` package was updated to use the new location in the same commit. No known external consumers of the old path.

- **[scope-creep]** — Acknowledged. The predicate model change is part of aligning with the real step action output, not incidental scope expansion.

### Low — addressed in d414a97b

- **[documentation-comment]** — Added doc comment on `intoto.subject_digest`. Fixed.

- **[missing-test]** — Added `test_missing_subject_triggers_mismatch` for the no-subject edge case. Fixed.

### Low — acknowledged, no change

- **[logic-error] `"n/a"` as count_key** — Follows `test.rego`'s `_did_result` pattern where ERROR and SKIPPED have no corresponding count field. The `"n/a"` sentinel makes `object.get(predicate, "n/a", 0)` return 0, so the count clause is a no-op. This is intentional — it keeps the helper signature uniform across all callers.

- **[missing-effective-on]** — Good point. Will discuss with @joejstuart whether phased rollout via `effective_on` is appropriate for the new deny rules.

File: policy/release/test_attestation/test_attestation.rego [hunk 1]
```diff
@@ -40,38 +43,109 @@ _test_name(statement) := name if {
...
 } else := "unknown test"
 
-_test_list(predicate, key) := result if {
-	value := object.get(predicate, key, [])
-	is_array(value)
-	items := [x | some x in value; is_string(x)]
-	count(items) > 0
-	result := concat(", ", items)
-} else := "(none listed)"
+_count_detail(predicate, key) := result if {
```
[the remark this answers is not in the document]
Comment (robnester-rh) [reply policy/release/test_attestation/test_attestation.rego:46]: Acknowledged — same behavior as test.rego. UX polish for a follow-up.

File: policy/release/test_attestation/test_attestation.rego [hunk 2]
```diff
@@ -131,28 +206,148 @@ deny contains result if {
...
 #   - redhat
 #   depends_on:
 #   - attestation_type.known_attestation_type
 #
-warn contains result if {
+deny contains result if {
 	some statement in _test_attestations
-	statement.predicate.result == "WARNED"
-	warned := _test_list(statement.predicate, "warnedTests")
+	_has_result(statement.predicate, rule_data.get("erred_test_attestation_results"), "n/a")
```
[the remark this answers is not in the document]
Comment (robnester-rh) [reply policy/release/test_attestation/test_attestation.rego:229]: Good point — added a comment to clarify the intent.

File: policy/lib/intoto/intoto.rego [hunk 3]
```diff
@@ -48,3 +48,13 @@ statements_by_predicate(predicate_type) := {statement |
...
 
 predicate_vuln_scan := "https://in-toto.io/attestation/vulns/v0.2"
+
+# Converts a subject digest map (e.g., {"sha256": "abc..."}) to a set of "algorithm:value" strings.
+subject_digests(subject) := {concat(":", [algorithm, value]) |
+	some algorithm, value in subject.digest
+}
+
+# Single-digest convenience accessor. Safe when subjects have one digest algorithm.
+subject_digest(subject) := digest if {
```
[the remark this answers is not in the document]
Comment (robnester-rh) [reply policy/lib/intoto/intoto.rego:58]: Acknowledged — the single-algorithm assumption is documented in the docstring. No action needed.

File: policy/release/test_attestation/test_attestation.rego [hunk 4]
```diff
@@ -40,38 +43,109 @@ _test_name(statement) := name if {
...
-} else := "(none listed)"
+_count_detail(predicate, key) := result if {
+	n := object.get(predicate, key, 0)
+	is_number(n)
+	n > 0
+	result := sprintf("%d", [n])
+} else := "0"
+
+_has_result(predicate, results, _) if {
+	predicate.result in {r | some r in results}
```
[the remark this answers is not in the document]
Comment (robnester-rh) [reply policy/release/test_attestation/test_attestation.rego:54]: No real-world regression — the `failedTests`/`warnedTests` string arrays were from a PoC example in the Jira, not from the actual `attest-test-result` step action. The step action has always produced integer counts (`failures`, `warnings`, `successes`). This change aligns the policy with the real producer.

**Comment by joejstuart** [comment]
I found an issue with the test attestation provenance discovery. I fixed it and rebased with this PR - https://github.com/conforma/policy/pull/1770
